File: phase-2/backend/models/task.py
from datetime import datetime
from typing import Optional
import uuid
from sqlmodel import Field, SQLModel


class TaskBase(SQLModel):
    title: str = Field(min_length=1, max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    completion_status: bool = Field(default=False)
    user_id: str  # UUID of the user from Better Auth


class Task(TaskBase, table=True):
    id: Optional[str] = Field(default_factory=lambda: str(uuid.uuid4()), primary_key=True, nullable=False)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    version: int = Field(default=1)  # For optimistic locking


class TaskCreate(SQLModel):
    title: str = Field(min_length=1, max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    completion_status: bool = Field(default=False)


class TaskRead(TaskBase):
    id: str
    created_at: datetime
    updated_at: datetime
    version: int


class TaskUpdate(SQLModel):
    title: Optional[str] = Field(default=None, min_length=1, max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    completion_status: Optional[bool] = None
    version: Optional[int] = None  # For optimistic locking

File: phase-4/frontend/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  completion_status: boolean;
  user_id: string;
  created_at: string;
  updated_at: string;
  priority: 'low' | 'medium' | 'high';  // New priority field
  tags: string[];  // New tags field
  due_date?: string;  // Existing due date field
}

export interface UserPreference {
  id: string;
  user_id: string;
  preference_key: string;
  preference_value: string;
  created_at: string;
  updated_at: string;
}